packages/core/test/text/textTests.tsx

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ * Licensed under the BSD-3 License as modified (the “License”); you may obtain a copy
+ * of the license at https://github.com/palantir/blueprint/blob/master/LICENSE
+ * and https://github.com/palantir/blueprint/blob/master/PATENTS
+ */
+
+import { assert } from "chai";
+import { mount } from "enzyme";
+import * as React from "react";
+
+import { Classes, Text } from "../../src/index";
+
+describe("<Text>", () => {
+    it("renders jsx text children", () => {
+        const children = (
+            <span>
+                {"computed text "}
+                <span>
+                    text in a span
+                </span>
+            </span>
+        );
+        const textContent = "computed text text in a span";
+        assert.lengthOf(document.getElementsByClassName(Classes.TEXT_OVERFLOW_ELLIPSIS), 0);
+
+        const wrapper = mount(<Text ellipsize>{children}</Text>);
+        const element = wrapper.find(`.${Classes.TEXT_OVERFLOW_ELLIPSIS}`);
+        assert.lengthOf(element, 1, `missing ${Classes.TEXT_OVERFLOW_ELLIPSIS}`);
+        assert.strictEqual(element.text(), textContent, "content incorrect value");
+    });
+});
```

Comment:
I know this is a simple component, but can we add more tests with clear descriptions communicating and verifying how this component behaves in various scenarios? An exhaustive list may look like:

```
describe("<Text>", () => {
    it("renders properly when content contains plain text");
    it("renders properly when content contains JSX elements");

    describe("if ellipsize=true", () => {
        it("truncates overflowing text to one line with trailing ellipses");
        it("does not attempt to truncate non-overflowing text");
        it("truncates text when it changes from non-overflowing to overflowing");
        it("un-truncates text when it changes from overflowing to non-overflowing");
    });

    describe("if ellipsize=false", () => {
        it("does not attempt to truncate overflowing text");
        it("does not attempt to truncate non-overflowing text");
        it("does not attempt to truncate text when it changes from non-overflowing to overflowing");
        it("does not attempt to truncate text when it changes from overflowing to non-overflowing");
    });
});
```